src/utils/helpers.py

```diff
@@ -0,0 +1,95 @@
+import csv
+import itertools
+import json
+from argparse import Namespace
+from pathlib import Path
+
+
+def construct_predicates(
+    csv_lines: list[list[str]], taxyr: int | list[int] | None = None
+) -> list[str]:
+    """
+    Constructs SQL predicates based on the provided CSV lines and tax year(s).
+
+    Args:
+        csv_lines: A list of start and end PARIDs used to divide a table. The
+            third value is the rough number of PARIDs in the batch.
+        taxyr: A single tax year, a list of tax years, or None. If provided,
+            will be appended to the PARID predicates. Defaults to None.
+
+    Returns:
+        A list of SQL predicate strings used to divide a table into chunks.
+    """
+    taxyr_list = [taxyr] if isinstance(taxyr, int) else taxyr
+    if taxyr_list:
+        predicates = [
+            f"parid >= '{start}' AND parid <= '{end}' AND taxyr = {year}"
+            for (start, end, _), year in itertools.product(
+                csv_lines, taxyr_list
+            )
+        ]
+    else:
+        predicates = [
+            f"parid >= '{start}' AND parid <= '{end}'"
+            for start, end, _ in csv_lines
+        ]
+
+    return predicates
+
+
+def read_predicates(path: str) -> list[list[str]]:
+    """
+    Read a CSV file containing PARID start and end boundaries.
+
+    Args:
+        path: String path to the CSV file, relative to the Docker container.
+
+    Returns:
+        A list of start and end PARIDs used to divide a table. The third value
+        is the rough number of PARIDs in the batch.
+    """
+    pred_path = Path(path)
+    with open(pred_path.resolve().as_posix(), mode="r") as file:
+        csv_reader = csv.reader(file)
+        next(csv_reader)
+        csv_lines = [row for row in csv_reader]
+
+    return csv_lines
+
+
+def strip_table_prefix(table_name: str) -> str:
+    """Removes the 'iasworld.' prefix from a table name."""
+    if table_name.startswith("iasworld."):
+        return table_name.split(".", 1)[1]
+    return table_name
+
+
+def load_job_config(args: Namespace) -> dict:
```

Comment:
**[Suggestion, non-blocking]** I have a preference for functions that explicitly name (and type hint) their arguments rather than accepting one object with an arbitrary number of attributes, since the latter design forces readers to read the implementation of the function in order to know which attributes the function ends up using. It's not a big deal though, and I think people disagree on this!

Reply:
Good call! Fixed in dcce2ac.